costmap_2d/src/costmap_2d_ros.cpp

```diff
@@ -359,7 +359,9 @@ void Costmap2DROS::reconfigureCB(costmap_2d::Costmap2DConfig &config, uint32_t l
 
   old_config_ = config;
 
-  map_update_thread_ = new boost::thread(boost::bind(&Costmap2DROS::mapUpdateLoop, this, map_update_frequency));
+  // only construct the thread if the frequency is positive
+  if(map_update_frequency > 0.0)
+    map_update_thread_ = new boost::thread(boost::bind(&Costmap2DROS::mapUpdateLoop, this, map_update_frequency));
```

Comment:
what happens if we previously had frequency > 0 and reconfigure to 0?

Reply:
We destroy the thread when we enter the reconfigureCB in lines 326-328. But I've missed to set the map_update_thread_ to NULL - fixed that with the newest commit

Reply:
Thanks!